core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -76,9 +76,9 @@ class MetadataCache(brokerId: Int) extends Logging {
         val leaderBrokerId = partitionState.basePartitionState.leader
         val leaderEpoch = partitionState.basePartitionState.leaderEpoch
         val maybeLeader = getAliveEndpoint(snapshot, leaderBrokerId, listenerName)
-        val replicas = partitionState.basePartitionState.replicas.asScala.map(_.toInt)
+        val replicas = partitionState.basePartitionState.replicas.asScala
```

Comment:
basically, these fields are all originally `j.u.List`s. There seems to be an unnecessary conversion to Scala collections and then back to Java (https://github.com/apache/kafka/pull/6397/files#diff-bfeebf48d90e86c1ffb850fbbe019dd6R112). I think we can avoid these altogether. Although there are Scala-collection-specific filters applied for debug logging (https://github.com/apache/kafka/pull/6397/files#diff-bfeebf48d90e86c1ffb850fbbe019dd6R107 for example), you can just use `JavaConverters.collectionAsScalaIterable` which IIUC does not do any copying.

Reply:
actually, nm - these aren't copying conversions except for the type conversion which your patch eliminates. So I think we are good.

Reply:
Yeah, none of the converters do copying. It's just unfortunate that there's no way to have `Int` as a collection parameter without copying. We can live with `Integer` here.